.github/workflows/ci.yml

```diff
@@ -0,0 +1,55 @@
+name: CI
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+  workflow_dispatch:
+
+jobs:
+  build_and_package:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-2019, macos-latest]
+
+    runs-on: ${{matrix.os}}
+    container: ${{ matrix.os == 'ubuntu-latest' && 'ghcr.io/f3d-app/f3d-ci' || null }}
+
+    env:
+      DISPLAY: :0
+
+    steps:
+
+    - name: Intall ninja Windows
+      if: matrix.os == 'windows-2019'
+      uses: seanmiddleditch/gha-setup-ninja@v3
+
+    - name: setup msvc Windows
+      if: matrix.os == 'windows-2019'
+      uses: TheMrMilchmann/setup-msvc-dev@v2
+      with:
+        arch: x64
+
+    - name: Checkout
+      uses: actions/checkout@v3
+      with:
+        path: 'source'
+        submodules: 'true'
+
+    - name: Setup Directories
+      working-directory: ${{github.workspace}}
+      run: mkdir build
+
+    - name: Configure
+      working-directory: ${{github.workspace}}/build
+      run: >
+        cmake ../source
+        -DCMAKE_BUILD_TYPE=Release
+        ${{ matrix.os == 'windows-2019' && '-G Ninja' || null }}
+
+    - name: Build
+      working-directory: ${{github.workspace}}/build
+      run: cmake --build . --parallel 2 --config Release
```

Comment:
The whole CI takes 2.5 hours to run, thats acceptable. We could consider caching the downloads but I dont think caching the builds would be a good idea.